docs/index.md

```diff
@@ -2602,10 +2602,63 @@ To build on our example above, here are some label filter queries and their beha
 | `ginkgo --label-filter="network && !slow"` | Run specs labelled `network` that aren't `slow` |
 | `ginkgo --label-filter=/library/` | Run specs with labels matching the regular expression `library` - this will match the three library-related specs in our example.
 
+##### Label Sets
+
+In addition to flat strings, Labels can also construct sets.  If a label has the format `KEY:VALUE` then a set with key `KEY` is created and the value `VALUE` is added to the set.  For example:
+
+```go
+Describe("The Library API", Label("API:Library"), func() {
+  It("can fetch a list of books", func() {
+    // has the labels [API:Library]
+    // API is a set with value {Library}
+  })
+  It("can fetch a list of books by shelf", Label("API:Shelf", "Readiness:Alpha"), func() {
+    // has the labels [API:Library, API:Shelf, Readiness:Alpha]
+    // API is a set with value {Library, Shelf}
+    // Readiness is a set with value {Alpha}
+
+  })
+  It("can fetch a list of books by zip code", Label("API:Geo", "Readiness:Beta"), func() {
+    // has the labels [API:Library, API:Geo, Readiness:Beta]
+    // API is a set with value {Library, Geo}
+    // Readiness is a set with value {Beta}
+  })
+})
+```
+
+Label filters can operate on sets using the notation: `KEY: SET_OPERATION <ARGUMENT>`. The following set operations are supported:
+
+| Set Operation | Argument | Description |
+| --- | --- | --- |
+| `isEmpty` | None | Matches if the set with key `KEY` is empty (i.e. no label of the form `KEY:*` exists) |
+| `containsAny` | `SINGLE_VALUE` or `{VALUE1, VALUE2, ...}` | Matches if the `KEY` set contains _any_ of the elements in `ARGUMENT` |
+| `containsAll` | `SINGLE_VALUE` or `{VALUE1, VALUE2, ...}` | Matches if the `KEY` set contains _all_ of the elements in `ARGUMENT` |
+| `consistsOf` | `SINGLE_VALUE` or `{VALUE1, VALUE2, ...}` | Matches if the `KEY` set contains _exactly_ the elements in `ARGUMENT` |
+| `isSubsetOf` | `SINGLE_VALUE` or `{VALUE1, VALUE2, ...}` | Matches if the elements in the `KEY` set are a subset of the elements in `ARGUMENT` |
+
+leading and trailing whitespace is alwasy trimmed around keys and values and comparisons are always case-insensitive.  Keys and values in the filter-language set operations are always literals; regular expressions are not supported.  A special note should be made about the behavior of `isSubsetOf`: if the `KEY` set is empty then the filter will always match.  This is because an empty set is always a subset of any other set.
```

Comment:
```suggestion
Leading and trailing whitespace is alwasy trimmed around keys and values and comparisons are always case-insensitive.  Keys and values in the filter-language set operations are always literals; regular expressions are not supported.

A special note should be made about the behavior of `isSubsetOf`: if the `KEY` set is empty then the filter will always match.  This is because an empty set is always a subset of any other set.
```